tracing-log/src/lib.rs

```diff
@@ -91,6 +93,70 @@ pub trait AsTrace {
     fn as_trace(&self) -> Self::Trace;
 }
 
+pub trait GenericLogEvent<'a> {
+    fn normalized_metadata(&self, fields: &'a LogVisitor) -> Option<Metadata<'a>>;
+    fn is_log(&self) -> bool;
+}
+
+impl<'a> GenericLogEvent<'a> for Event<'a> {
+    fn normalized_metadata(&self, fields: &'a LogVisitor) -> Option<Metadata<'a>> {
```

Comment:
It's a shame that the visitor has to be passed in (and that the `LogVisitor` type has to be public) but that's also a sad consequence of the erased lifetime in the `Visit` trait. Hmm...